client/src/store/slices/mentor/mentorSlice.ts

```diff
@@ -0,0 +1,160 @@
+/*
+This software is Copyright ©️ 2020 The University of Southern California. All Rights Reserved. 
+Permission to use, copy, modify, and distribute this software and its documentation for educational, research and non-profit purposes, without fee, and without a written agreement is hereby granted, provided that the above copyright notice and subject to the full license file found in the root of this software deliverable. Permission to make commercial use of this software may be obtained by contacting:  USC Stevens Center for Innovation University of Southern California 1150 S. Olive Street, Suite 2300, Los Angeles, CA 90115, USA Email: accounting@stevens.usc.edu
+
+The full terms of this copyright and license should always be found in the root directory of this software deliverable as "license.txt" and if these terms are not found with this software, please contact the USC Stevens Center for the full license.
+*/
+import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
+import * as api from "api";
+import { LoadingError } from "hooks/graphql/loading-reducer";
+import { Mentor } from "types";
+
+/** Store */
+
+export enum MentorStatus {
+  NONE = 0,
+  LOADING = 1,
+  SAVING = 2,
+  SUCCEEDED = 3,
+  FAILED = 4,
+}
+
+export interface MentorState {
+  data: Mentor | undefined;
+  editedData: Mentor | undefined;
+  isEdited: boolean;
+  mentorStatus: MentorStatus;
+  error: LoadingError | undefined;
+  accessToken: string;
+}
+
+export interface Headers {
+  accessToken: string;
+  editedData: Mentor | undefined;
+}
+
+const initialState: MentorState = {
+  data: undefined,
+  editedData: undefined,
+  isEdited: false,
+  mentorStatus: MentorStatus.NONE,
+  error: undefined,
+  accessToken: "",
+};
+
+/** Actions */
+
+export const getMentor = createAsyncThunk(
```

Comment:
redux-toolkit is new to me: do we not need an explicit type on consts created by `createAsyncThunk` because typrescript infers the type? Does vscode show the variable as fully typed on mouseover?

Reply:
yeah, it seems like it